src/Engine/Component/SkinningComponent.cpp

```diff
@@ -0,0 +1,497 @@
+    // get the current animation data.
+    bool hasSkel    = compMsg->canGet<Skeleton>( getEntity(), m_skelName );
+    bool hasRefPose = compMsg->canGet<RefPose>( getEntity(), m_skelName );
+    bool hasMesh    = compMsg->canGet<TriangleMesh>( getEntity(), m_meshName );
+
+    if ( hasSkel && hasMesh && hasRefPose )
+    {
+        m_renderObjectReader =
+            compMsg->getterCallback<Ra::Core::Utils::Index>( getEntity(), m_meshName );
+        m_skeletonGetter = compMsg->getterCallback<Skeleton>( getEntity(), m_skelName );
+        m_meshWriter     = compMsg->rwCallback<TriangleMesh>( getEntity(), m_meshName );
+
+        // copy mesh triangles and find duplicates for normal computation.
+        TriangleMesh* mesh = const_cast<TriangleMesh*>( m_meshWriter() );
+        m_refData.m_referenceMesh.copy( *mesh );
+        findDuplicates( *mesh, m_duplicatesMap );
+
+        // get other data
+        m_refData.m_skeleton = compMsg->get<Skeleton>( getEntity(), m_skelName );
+        createWeightMatrix();
+        m_refData.m_refPose = compMsg->get<RefPose>( getEntity(), m_skelName );
+        applyBindMatrices( m_refData.m_refPose );
+
+        // initialize frame data
+        m_frameData.m_frameCounter = 0;
+        m_frameData.m_doSkinning   = true;
+        m_frameData.m_doReset      = false;
+
+        m_frameData.m_previousPos   = m_refData.m_referenceMesh.vertices();
+        m_frameData.m_currentPos    = m_refData.m_referenceMesh.vertices();
+        m_frameData.m_currentNormal = m_refData.m_referenceMesh.normals();
+
+        m_frameData.m_previousPose = m_refData.m_refPose;
+        m_frameData.m_currentPose  = m_refData.m_refPose;
+        m_frameData.m_refToCurrentRelPose =
+            Ra::Core::Animation::relativePose( m_frameData.m_currentPose, m_refData.m_refPose );
+        m_frameData.m_prevToCurrentRelPose = Ra::Core::Animation::relativePose(
+            m_frameData.m_currentPose, m_frameData.m_previousPose );
+
+        // setup comp data
+        m_isReady     = true;
+        m_forceUpdate = true;
+        setupSkinningType( m_skinningType );
+        setupSkinningType( STBS_LBS ); // ensure weights are present for display
+
+        // prepare RO for skinning weights display
+        auto ro         = getRoMgr()->getRenderObject( *m_renderObjectReader() );
+        m_baseTechnique = ro->getRenderTechnique();
+
+        auto attrUV = Ra::Engine::Mesh::getAttribName( Ra::Engine::Mesh::VERTEX_TEXCOORD );
+        if ( mesh->hasAttrib( attrUV ) )
+        {
+            auto handle = mesh->getAttribHandle<Ra::Core::Vector3>( attrUV );
+            m_baseUV    = mesh->getAttrib( handle ).data();
+        }
+
+        m_weightTechnique.reset( new Ra::Engine::RenderTechnique );
+        auto builder = Ra::Engine::EngineRenderTechniques::getDefaultTechnique( "BlinnPhong" );
+        builder.second( *m_weightTechnique.get(), true );
+        auto matT = std::static_pointer_cast<Ra::Engine::BlinnPhongMaterial>(
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Variable 'matT' is assigned a value that is never used.](https://app.codacy.com/manual/STORM/Radium-Engine/pullRequest?prid=5394305)